streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala

```diff
@@ -140,6 +141,16 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
     }
     true
   }
+
+  test("registering listeners via spark.streaming.extraListeners") {
+    val conf = new SparkConf().setMaster("local").setAppName("test")
+      .set("spark.streaming.extraListeners", classOf[BatchInfoCollector].getName + "," +
```

Comment:
I missed it earlier, this unit does not cover the two cases, one without constructor and one with constructor.
It may be easier to read if you create new classes - `StreamingListenerWithoutArg`, and `StreamingListenerWithConf`
(similar to ListenerThatAcceptsSparkConf for spark.extraListeners in SparkListenerSuite)


Also, the `StreamingContextSuite` is getting really overloaded, mind shifting this to `StreamingListenerSuite`, and reverting this file?